File: JAVA/ABSTRAÇAO/funcionario.java
package JAVA.ABSTRAÇAO;

public abstract class Funcionario {
        String nome;
        double salario;


    public Funcionario(String nome ,double salario){
        this.nome = nome;
        this.salario = salario;
    } 

    public  abstract void  trabalhar();
    public abstract void sair();
}

File: JAVASCRIPT/ABSTRAÇAO/funcionario.js


// ABSTRAÇÃO: Classe base que define o modelo genérico e esconde detalhes complexos
class Funcionario {
    constructor(nome, salarioBase) {
        if (this.constructor === Funcionario) {
            throw new Error("Classes abstratas não podem ser instanciadas diretamente!");
        }
        this.nome = nome;
        this.salarioBase = salarioBase;
    }

    // Método abstrato que força as classes filhas a implementarem
    calcularBonus() {
        throw new Error("O método calcularBonus precisa ser implementado na subclasse.");
    }
}

module.exports = Funcionario;



















